### PR TITLE
tests: fix flaky cache test

### LINKED ISSUE #11772: flaky test in `uses-long-cache-ttl-test.js`
`Cache headers audit › respects expires/cache-control priority`

image

### PR DESCRIPTION
**Pull request by connorjclark**
fixes #11772

### CODE REVIEW

File: lighthouse-core/test/audits/byte-efficiency/uses-long-cache-ttl-test.js [hunk 1]
```diff
@@ -116,6 +116,11 @@ describe('Cache headers audit', () => {
   });
 
   it('respects expires/cache-control priority', () => {
+    // Stub Date.now so the test is not sensitive to timing.
+    const now = Date.now();
```
Comment (connorjclark) [inline lighthouse-core/test/audits/byte-efficiency/uses-long-cache-ttl-test.js:120]: could make the argument that this should be in a beforeAll. thoughts?

Comment (patrickhulce) [reply lighthouse-core/test/audits/byte-efficiency/uses-long-cache-ttl-test.js:120]: I would make that argument yes :)

**Review by patrickhulce — CHANGES_REQUESTED** [review]

**Review by patrickhulce — APPROVED** [review]

**Comment by patrickhulce** [comment]
wait hang on, how does this fix the flake...? isn't the problem that too much time elapses between `now` and the assertion which is still true?

**Comment by connorjclark** [comment]
how does time pass if date.now is constant. sppoky time

**Comment by brendankenny** [comment]
so no to switching the audit off of `Date` and to request timing?

**Comment by brendankenny** [comment]
it seems like from

https://github.com/GoogleChrome/lighthouse/blob/a938fc497a710c723fb21a70460d38a03ae88fe6/lighthouse-core/audits/byte-efficiency/uses-long-cache-ttl.js#L118-L124

auditing with `-A` would often mean any headers hitting this branch would always have a negative `cacheLifetimeInSeconds`, which we then discard down in 

https://github.com/GoogleChrome/lighthouse/blob/a938fc497a710c723fb21a70460d38a03ae88fe6/lighthouse-core/audits/byte-efficiency/uses-long-cache-ttl.js#L222-L228

so would never show up in the audit?

I realize this is expanding the scope of the intended fix, so this can also be a new issue :)

**Comment by connorjclark** [comment]
merging in interest of less flaky tests. i'll write a new issue but can't take it right away.

**Pull request merged**